backend/src/main/java/com/google/blackswan/mock/SimpleRelatedDataGenerator.java

```diff
@@ -0,0 +1,119 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.blackswan.mock;
+
+import com.google.models.*;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.ArrayListMultimap;
+
+/** Singleton class to generate related data for a given anomaly. */
+public class SimpleRelatedDataGenerator implements RelatedDataGenerator {
+  private static final String TARGET_DIMENSION = "Ramen";
+  private static final String RELATED_METRIC = "Interest Over Time";
+  private static final String CONFIG_USERNAME = "catyu@";
+  private static final ImmutableMap<String, String> FILE_LOCATION = ImmutableMap.of(
+                                                                      "Udon", "/udon-data.csv",
+                                                                      "Pho", "/pho-data.csv"
+                                                                    );
+
+  private static final SimpleRelatedDataGenerator INSTANCE = new SimpleRelatedDataGenerator();
+
+  private Multimap<String, DataInfo> relatedDataMap;
+  private Map<String, Map<Timestamp, Integer>> csvDataCache;
+
+  private SimpleRelatedDataGenerator() {
+    relatedDataMap = ArrayListMultimap.create();
+    csvDataCache = new HashMap<>();
+    prefillRelatedData(); // TODO: Remove when connected to datastore.
+  }
+
+  /** 
+   * Simulates action of querying configs from datastore and adding to 
+   * relatedDataMap. Currently, manually adds these relationship.
+   */
+  private void prefillRelatedData() {
+    // TODO: Replace with call for querying configs from datastore.
+    relatedDataMap.put("Ramen", new DataInfo(RELATED_METRIC, "Udon", CONFIG_USERNAME));
+    relatedDataMap.put("Ramen", new DataInfo(RELATED_METRIC, "Pho", CONFIG_USERNAME));
+    // Multimap looks like this right now: {{Ramen,{Udon, Pho}}, ...}.
+  }
+
+  public List<RelatedData> getRelatedData(String metricName, String dimensionName, 
+      Timestamp startTime, Timestamp endTime) {
+    List<RelatedData> relatedDataList = new ArrayList<RelatedData>();
+    // TODO: Currently only tries to find related data for dimension, ex. Ramen -> Pho, 
+    //       does not include logic for metric, should be similar.
+    if (relatedDataMap.containsKey(dimensionName)) {
```

Comment:
what happens if the map doesn't contain the dimension name the user wants to get related data for?

Reply:
then the `relatedDataList` will simply be an empty list! is that okay?

Reply:
Yeah I guess that's okay. Maybe write a comment that explains this behavior (If dimension name not found, returns an empty list.) But if you're changing this code later on it's no big deal